src/Symfony/Component/Semaphore/Store/Resources/redis_put_off_expiration.lua

```diff
@@ -9,10 +9,12 @@ if added == 1 then
 end
 
 -- Extend the TTL
-local curentTtl = redis.call("TTL", weightKey)
-if curentTtl < now + ttlInSecond then
-    redis.call("EXPIRE", weightKey, curentTtl + 10)
-    redis.call("EXPIRE", timeKey, curentTtl + 10)
+local maxExpiration = redis.call("ZREVRANGE", timeKey, 0, 0, "WITHSCORES")[2]
```

Comment:
@jderusse I reverted the change you asked me. Because it does not work

Reply:
thanks for notifying, I probably have to fix it in my PR.

What's the issue?

Reply:
All keys are released too early and so nothing works 😬